app/logical/sources/alternates/furaffinity.rb

```diff
@@ -1,7 +1,7 @@
 module Sources
   module Alternates
     class Furaffinity < Base
-      IMAGE_TO_ARTIST = /facdn\.net\/art\/([0-9a-zA-Z_.~\-]+)/
+      IMAGE_TO_ARTIST = /d2?\.(?!facdn|furaffinity)\.net\/art\/([0-9a-zA-Z_.~\-\[\]]+)/
```

Comment:
This is using a negative lookahead (`?!`) rather than the non capturing group I assume you're going for (`?:`), so this will never match